### PR TITLE
Backup/Restore Suite - Recursefix

### LINKED ISSUE #7387: How should restore-dbadatabase work with subfolders in -path
From the help for restore-dbadatabase (https://docs.dbatools.io/Restore-DbaDatabase) it says this:
-Path
Path to SQL Server backup files. Paths passed in as strings will be scanned using the desired method, default is a non recursive folder scan Accepts multiple paths separated by ',' Or it can consist of FileInfo objects, such as the output of Get-ChildItem or Get-Item. This allows you to work with your own file structures as needed

With the testing I am doing, I am seeing that it IS recursive.  I have full backups in a subfolder and tlog backups in the parent folder (i.e. d:\backups and d:\backups\fulls) and it is scanning both folders in get-dbabackupinformation and restore-dbadatabase.  For the -path parameter, I only specify d:\backups and it is restoring the full backup from d:\backups\fulls.  I do not understand why it is doing it when it says the default is non-recursive.  Why is it scanning and restoring from d:\backups\Fulls when I only specify d:\backups?

I can deal with it either way.  I just want to understand how it is programmed to work and does the help align with what it is doing?

I am using version 1.0.145 of dbatools and powershell version 5.

Thanks!

### PR DESCRIPTION
**Pull request by Stuart-Moore**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #7387)
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Unit test is included
 - [x] Documentation
 - [ ] Build system

### Purpose
Should make the Restore-Dbadatabase documentation match up with reality

The appv failures don't appear to have anything to with this

### CODE REVIEW

**Comment by Stuart-Moore** [comment]
@lauraberanek could you read the new documentation please, and let me know if it reads as how the command actually works

File: functions/Get-DbaBackupInformation.ps1 [hunk 1]
```diff
@@ -144,6 +147,8 @@ function Get-DbaBackupInformation {
         [parameter(ParameterSetName = "Create")]
         [Switch]$NoXpDirTree,
         [parameter(ParameterSetName = "Create")]
+        [Switch]$NoXpDirRecurse = $false,
```
Comment (andreasjordan) [inline functions/Get-DbaBackupInformation.ps1:150]: Isn't $false always the default for a swithch?

File: functions/Get-DbaBackupInformation.ps1 [hunk 2]
```diff
@@ -39,7 +39,10 @@ function Get-DbaBackupInformation {
         If provided only backup originating from this destination will be returned. This SQL instance will not be connected to or involved in this work
 
     .PARAMETER NoXpDirTree
-        If this switch is set, then Files will be parsed as locally files. This can cause failures if the running user can see files that the parsing SQL Instance cannot
+        If this switch is set, then Files will be parsed as local files. This can cause failures if the running user can see files that the parsing SQL Instance cannot
```
Comment (wsmelton) [inline functions/Get-DbaBackupInformation.ps1:42]:
```suggestion
        If specified, this switch will cause the files to be parsed as local files to the SQL Server Instance provided. Errors may be observed when the SQL Server Instance cannot access the files being parsed.
```

This wording may bring the "errors" in better view of the reader.

File: functions/Get-DbaBackupInformation.ps1 [hunk 3]
```diff
@@ -39,7 +39,10 @@ function Get-DbaBackupInformation {
...
+
+    .PARAMETER NoXpDirRecurse
+        If this switch is set, then xp_dirtree will not recurse down folders
```
Comment (wsmelton) [inline functions/Get-DbaBackupInformation.ps1:45]:
```suggestion
        If specified, this switch changes xp_dirtree behavior to not recurse the folder structure.
```

File: functions/Get-DbaBackupInformation.ps1 (hunk 1 above)
Comment (potatoqualitee) [reply functions/Get-DbaBackupInformation.ps1:150]: yes, this does seem redundant

**Comment by potatoqualitee** [comment]
will wait for explicit approval or/ stuart you can merge when it's ready

**Pull request merged**